src/NoteMouseHandler/NoteMouseHandler.js

```diff
@@ -36,62 +34,54 @@ export default class MouseHandler {
       registerMouseDown,
       registerMouseMove,
       registerMouseUp)
-    actionMouseDown(e, ctx)
+    actionMouseDown(local)
   }
 
-  onMouseMove(e, ctx) {
+  onMouseMove(local, ctx, e) {
     if (this.action) {
-      this.actionMouseMove(e, ctx)
+      this.actionMouseMove(local, e)
     } else {
       let cursor = "auto"
       if (this.cursorHandler) {
-        cursor = this.cursorHandler(e, ctx)
+        cursor = this.cursorHandler(local, ctx)
       }
-      this.emitter.trigger("change-cursor", cursor)
+      ctx.changeCursor(cursor)
     }
   }
 
-  onMouseUp(e, ctx) {
+  onMouseUp(local) {
     if (this.action) {
-      this.actionMouseUp(e, ctx)
+      this.actionMouseUp(local)
     }
     this.action = null
   }
 }
 
-export function defaultActionFactory(e) {
+export function defaultActionFactory(local, ctx, e) {
   if (e.nativeEvent.button == 1) {
     // wheel drag to start scrolling
     return dragScrollAction
   }
 
   if (e.nativeEvent.button == 2 && e.nativeEvent.detail == 2) {
-    return changeToolAction
+    return changeToolAction(ctx)
   }
 
   return null
 }
 
-function dragScrollAction(emitter, onMouseDown, onMouseMove) {
-  onMouseMove(e => {
-    emitter.trigger("drag-scroll", {movement: {
-      x: e.nativeEvent.movementX,
-      y: e.nativeEvent.movementY
-    }})
+function dragScrollAction(onMouseDown, onMouseMove) {
+  onMouseMove((local, e) => {
+    // FIXME
+    // emitter.trigger("drag-scroll", {movement: {
+    //   x: e.nativeEvent.movementX,
+    //   y: e.nativeEvent.movementY
+    // }})
   })
 }
 
-function changeToolAction(emitter, onMouseDown) {
+const changeToolAction = ctx => onMouseDown => {
```

Comment:
__[jshint]__
*  'const' is available in ES6 (use 'esversion
*  'arrow function syntax (=>)' is only available in ES6 (use 'esversion

<!-- Comment by [SideCI](https://sideci.com) -->